circuits/src/zk_circuits/valid_wallet_create.rs

```diff
@@ -301,107 +286,136 @@ where
-
-        // Absorb the wallet randomness into the hasher state
-        arkworks_hasher.absorb(&scalar_to_prime_field(&witness.wallet_randomness));
-
-        prime_field_to_scalar::<DalekRistrettoField>(
-            &arkworks_hasher.squeeze_field_elements(1 /* num_elements */)[0],
-        )
+        (witness, statement)
     }
 
-    /// Tests the VALID WALLET CREATE with a valid witness (empty wallet)
-    #[test]
-    fn test_valid_wallet() {
+    /// Asserts that a given witness, statement pair is invalid
+    fn assert_invalid_witness_statement(witness: SizedWitness, statement: SizedStatement) {
+        // Create a constraint system
+        let pc_gens = PedersenGens::default();
+        let mut transcript = Transcript::new(b"test");
+        let mut prover = Prover::new(&pc_gens, &mut transcript);
+
+        // Allocate the witness and statement in the constraint system
         let mut rng = OsRng {};
-        let n_fees = MAX_FEES;
-        let fees = (0..n_fees).map(|_| random_fee(&mut rng)).collect_vec();
+        let (witness_var, _) = witness.commit_witness(&mut rng, &mut prover).unwrap();
+        let statement_var = statement.commit_public(&mut prover).unwrap();
 
-        let witness = ValidWalletCreateWitness {
-            fees: fees.try_into().unwrap(),
-            keys: PUBLIC_KEYS.clone(),
-            wallet_randomness: Scalar::random(&mut rng),
-        };
-        let statement = ValidWalletCreateStatement {
-            wallet_commitment: compute_commitment(&witness),
-        };
+        // Apply the constraints
+        ValidWalletCreate::circuit(statement_var, witness_var, &mut prover).unwrap();
+        assert!(!prover.constraints_satisfied());
+    }
+
+    // ---------
+    // | Tests |
+    // ---------
+
+    /// Tests that the circuit correctly verifies with valid zero'd balance and orders lists
+    #[test]
+    fn test_valid_initial_wallet() {
+        let (witness, statement) = create_default_witness_statement();
 
-        // Prove and verify on a smaller (for testing speed) version of the circuit
         let res = bulletproof_prove_and_verify::<
             ValidWalletCreate<MAX_BALANCES, MAX_ORDERS, MAX_FEES>,
         >(witness, statement);
-        assert!(res.is_ok());
+        assert!(res.is_ok())
+    }
+
+    /// Tests the case in which the commitment to the private shares is incorrect
+    #[test]
+    fn test_invalid_commitment() {
+        let (witness, mut statement) = create_default_witness_statement();
+        statement.private_shares_commitment += Scalar::from(1u8);
+
+        assert_invalid_witness_statement(witness, statement);
+    }
+
+    /// Tests the case in which a non-zero order is given
+    #[test]
+    fn test_nonzero_order() {
+        let mut wallet = create_empty_wallet();
+        wallet.orders[0] = INITIAL_ORDERS[0].clone();
+
+        let (witness, statement) = create_witness_statement_from_wallet(&wallet);
```

Comment:
i may be missing something obvious, but why should this test (and the one below) fail? yeah, we fudge the orders to be non-zero, but we create a witness statement from the wallet after having done this, and afaict `create_witness_statement_from_wallet` makes no assumptions about the wallet thats passed in being an empty wallet.

perhaps in this line we should call `create_default_witness_statment`?

Reply:
We want `create_witness_statement_from_wallet` to create an invalid witness; the fact that it makes no assumptions *is* the desired behavior. This test case is testing that the constraints are not satisfied for an invalid witness; and `create_default_witness_statement` creates a *valid* witness.